src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/codeSmell/MagicNumberInspector.java

```diff
@@ -0,0 +1,290 @@
+                }
+
+                return false;
+            }
+
+            private boolean isNumeric(final PsiElement expression) {
+                PsiElement testingExpression = expression;
+
+                if (testingExpression instanceof UnaryExpression) {
+                    testingExpression = ((UnaryExpression) testingExpression).getValue();
+                }
+
+                return (testingExpression instanceof PhpExpressionImpl) &&
+                       (testingExpression.getNode().getElementType() == PhpElementTypes.NUMBER);
+            }
+
+            private boolean isNotZeroNumber(@Nullable final PsiElement value) {
+                return (value != null) &&
+                       isNumeric(value) &&
+                       !"0".equals(value.getText());
+            }
+
+            private boolean isNumberOneNegative(final PsiElement unaryExpression) {
+                if (!(unaryExpression instanceof UnaryExpression)) {
+                    return false;
+                }
+
+                final PhpPsiElement unaryValue = ((UnaryExpression) unaryExpression).getValue();
+
+                return (unaryValue != null) &&
+                       "1".equals(unaryValue.getText());
+            }
+
+            private boolean isCounting(@NotNull final BinaryExpression binaryExpression) {
+                final IElementType operationType = binaryExpression.getOperationType();
+
+                if (operationType == null) {
+                    return false;
+                }
+
+                final boolean    numericOnOpposite = isNumeric(binaryExpression.getRightOperand());
+                final PsiElement numericOperand    = numericOnOpposite ? binaryExpression.getRightOperand() : binaryExpression.getLeftOperand();
+
+                if (numericOperand == null) {
+                    return false;
+                }
+
+                if (operationType.equals(PhpTokenTypes.opIDENTICAL) ||
+                    operationType.equals(PhpTokenTypes.opNOT_IDENTICAL) ||
+                    operationType.equals(PhpTokenTypes.opEQUAL) ||
+                    operationType.equals(PhpTokenTypes.opNOT_EQUAL)) {
+                    return "0".equals(numericOperand.getText());
+                }
+
+                final IElementType normalizeOperationType = numericOnOpposite ? operationType : ElementTypeUtil.rotateOperation(operationType);
+
+                return (normalizeOperationType.equals(PhpTokenTypes.opGREATER) ||
+                        normalizeOperationType.equals(PhpTokenTypes.opGREATER_OR_EQUAL)) &&
+                       isBinaryNumeric(numericOperand.getText());
+
+            }
+
+            private boolean isBinaryNumeric(@NotNull final String numericValue) {
+                return "0".equals(numericValue) ||
+                       "1".equals(numericValue);
+            }
+
+            private void registerProblem(final PsiElement rightOperand) {
```

Comment:
I did that to centralize the responsability at one method, once that we have a lot of calls like that. I will change too.